aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -84,6 +118,19 @@ bool is_square_or_vec(int64_t dim_i, int64_t dim_j, int64_t dim_k) {
   return (dim_i == dim_k  && dim_k == dim_j) || (dim_i == dim_j && dim_k == 1);
 }
 
+Tensor& sin_sparse_csr_out(const Tensor& self, Tensor& result) {
+  return unary_op_out(&at::sin_outf, self, result);
+}
+
+Tensor sin_sparse_csr(const Tensor& self) {
+  auto result = get_result_tensor_for_unary_op(&at::sin, self);
+  return sin_sparse_csr_out(self, result);
+}
+
+Tensor& sin_sparse_csr_(Tensor& self) {
+  return sin_sparse_csr_out(self, self);
+}
```

Comment:
I'm thinking of a macro-based approach for this, something similar to what's done here: https://github.com/pytorch/pytorch/blob/e795315c638228d4170f3797356c09a70b2ed4cd/aten/src/ATen/native/UnaryOps.cpp#L138-L143

But for now, I think this is okay, and we can look at it again in the next few weeks. Open to opinions and suggestions. :)

Reply:
For now, current explicit approach is good enough, once we have 2+ functions exploring macros might be worth it.